src/history/hs_rec.c

```diff
@@ -817,6 +824,29 @@ __hs_delete_key_from_ts_int(
     return (ret);
 }
 
+/*
+ * __wt_hs_delete_key_from_ts_btree --
+ *     Delete history store content of a given key from a timestamp.
+ */
+static int
+__wt_hs_delete_key_from_ts_btree(
+  WT_SESSION_IMPL *session, uint32_t btree_id, const WT_ITEM *key, wt_timestamp_t ts, bool reinsert)
+{
+    WT_DECL_RET;
+
+    /* The session should be pointing at the history store btree. */
+    WT_ASSERT(session, WT_IS_HS(S2BT(session)));
```

Comment:
Could we copy this assertion to `__hs_delete_key_from_ts_btree_int`?

Reply:
fixed.